rhinosystem/window.py

```diff
@@ -31,20 +30,29 @@ class RhinosystemWindow(Adw.ApplicationWindow):
     __gtype_name__ = 'RhinosystemWindow'
 
     stack_view: Gtk.Stack = Gtk.Template.Child()
-    version: Gtk.Label = Gtk.Template.Child()
+    version: Gtk.Button = Gtk.Template.Child()
+    version_invalid: Gtk.Button = Gtk.Template.Child()
     title: Gtk.Label = Gtk.Template.Child()
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self.os_info = SysinfoView()
         self.upgrade_progress = UpgradeView()
+        self.version.connect("clicked", self.copy_version)
         self.os_info.upgrade_button.connect("clicked", self.upgrade_os)
         self.stack_view.add_child(self.os_info)
         self.stack_view.add_child(self.upgrade_progress)
-        self.version.set_label(DeviceInfo().get_os_version())
+        if DeviceInfo().get_os_version() != "Unknown":
+            validation: float = float(DeviceInfo().get_os_version())
+            self.version_invalid.set_visible(False)
+            self.version.set_label(DeviceInfo().get_os_version())
+        else: self.version.set_visible(False)
 
     def upgrade_os(self, widget):
         self.os_info.set_visible(False)
         self.upgrade_progress.set_visible(True)
         self.upgrade_progress.on_show(self)
         self.title.set_label("Updating Rhino Linux")
+
+    def copy_version(self, widget):
+        system(f"echo -n \"{DeviceInfo().get_os_version()}\" | xclip -selection c")
```

Comment:
This will not work on any Wayland compositors, a global way of copying will be better.
Additionally you shouldn't use os.system, but the wrapper function for commands in https://github.com/rhino-linux/rhino-system/blob/main/rhinosystem/utils/command.py

Reply:
My initial plan was to use pyperclip, however it unfortunately doesn't work. I can try to find a global solution that doesn't use xclip. However for now, this code will work for users of Unicorn (XFCE) and other Xorg Desktops. Once a global solution for both X and wayland is found, it'll be implimented.

Reply:
Checking if wayland or xorg is being used is possible, and the command can be changed based on that.